### PR TITLE
KRKNWK:16512-sample.yaml adjusted to build all samples needed for nightly tests

### PR DESCRIPTION
**Pull request by HubertHugo**

### CODE REVIEW

File: samples/matter/lock/sample.yaml [hunk 1]
```diff
@@ -60,3 +60,25 @@ tests:
     integration_platforms:
       - nrf52840dk_nrf52840
     platform_allow: nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp
+  sample.matter.lock.release.ffs:
```
Comment (jnsgsr) [inline samples/matter/lock/sample.yaml:63]: Add all new configuration to quarantine so they will not build in integration jobs, that is move them above the comment line (line 35 here), and put them to scripts/quarantine.yml file.

File: samples/matter/lock/sample.yaml [hunk 2]
```diff
@@ -60,3 +60,25 @@ tests:
...
+    build_only: true
+    extra_args: CONF_FILE=prj_release.conf CONFIG_CHIP_COMMISSIONABLE_DEVICE_TYPE=y
+      CONFIG_CHIP_ROTATING_DEVICE_ID=y CONFIG_CHIP_DEVICE_TYPE=10
+    platform_allow: nrf52840dk_nrf52840
```
Comment (jnsgsr) [inline samples/matter/lock/sample.yaml:67]: I don't think that is true, it should be allowed for more platforms.

File: samples/matter/lock/sample.yaml (hunk 1 above)
Comment (jnsgsr) [inline samples/matter/lock/sample.yaml:63]: Use integration platforms argument to choose for which platform it should build, here and for other configurations where integration_platforms is missing.

**Review by jnsgsr — CHANGES_REQUESTED** [review]

File: samples/matter/lock/sample.yaml [hunk 3]
```diff
@@ -60,3 +82,11 @@ tests:
     integration_platforms:
       - nrf52840dk_nrf52840
     platform_allow: nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp
+  sample.matter.lock.factory_data:
+    build_only: true
+    extra_args: CONFIG_CHIP_FACTORY_DATA=y -CONFIG_CHIP_FACTORY_DATA=y -CONFIG_CHIP_FACTORY_DATA_BUILD=y
```
Comment (jnsgsr) [inline samples/matter/lock/sample.yaml:87]: `CONFIG_CHIP_FACTORY_DATA=y` duplicated, and there are redundant `-`

**Review by jnsgsr — CHANGES_REQUESTED** [review]

**Review by jnsgsr — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

File: scripts/requirements-fixed.txt [hunk 4]
```diff
@@ -38,6 +38,7 @@ intelhex==2.3.0
 intervaltree==3.1.0
 isort==5.7.0
 Jinja2==2.11.3
+jsonschema==4.4.0
```
Comment (tejlmand) [inline scripts/requirements-fixed.txt:41]: Late to the game, but https://github.com/nrfconnect/sdk-nrf-next/pull/1031 brought my attention to this change.

Why is `jsonshema` suddenly placed here.
afact, we don't have any `jsonshema` pip requirement in our regular `requirements-*.txt` files.

Is this requirement only for CI ?
We should know why a given pip package is required, and in this case we try to aling to yaml, so I would really know why json is needed.

Also this file serves as input to NCS Toolchain creation, so we shouldn't hace pip packages described here if they are only needed for CI testing.
If it is required for general development, we need an entry somewhere in a requirements file.

**Review by tejlmand — COMMENTED** [review]
Late to the game, but we need to have a talk on this update.

File: scripts/requirements-fixed.txt (hunk 4 above)
Comment (jnsgsr) [reply scripts/requirements-fixed.txt:41]: @LuDuda @ArekBalysNordic FYI

Comment (jnsgsr) [reply scripts/requirements-fixed.txt:41]: @tejlmand as this requirement is not "only for CI", and it is to update the build system environment for Matter sample accessories, how exactly should I correct it, or who can decide on this?

**Pull request merged**